### ISSUE TITLE
failed to obtain hg38

### ISSUE THREAD

**Issue by cmatKhan** [lawremi/rtracklayer#56]
This works if I use hg19, but not hg38 -- is there a different way to use hg38?

```
> targetTrack = with(targets_df, GRangesForUCSCGenome("hg38", "chr16", ranges, "+", "CDH13"))

Error in GRangesForGenome(genome, chrom = chrom, ranges = ranges, method = "UCSC",  : 
  Failed to obtain information for genome 'hg38'
  ```

**Comment by cmatKhan** [comment]
I posted this on the bioconductor support page, also. There was a request for more details, which I can appreciate:

I am trying to follow the rtracklayer vignette. The object setup is everything before section 2.2, but it is section 2.2 which describes how to launch the UCSC browser.

The annotations themselves are imported using rtracklayer::import() from gencode38.

here is what each of the objects look like:

```
> targets_df
DataFrame with 1 row and 21 columns
                     source     type     score     phase            gene_id      gene_type   gene_name       level     hgnc_id
                   <factor> <factor> <numeric> <integer>        <character>    <character> <character> <character> <character>
ENSG00000140945.17   HAVANA     gene        NA        NA ENSG00000140945.17 protein_coding       CDH13           1   HGNC:1753
                            havana_gene transcript_id transcript_type transcript_name transcript_support_level               tag
                            <character>   <character>     <character>     <character>              <character>       <character>
ENSG00000140945.17 OTTHUMG00000176635.3            NA              NA              NA                       NA overlapping_locus
                   havana_transcript exon_number     exon_id         ont  protein_id      ccdsid
                         <character> <character> <character> <character> <character> <character>
ENSG00000140945.17                NA          NA          NA          NA          NA          NA
```

```
> chr
factor-Rle of length 1 with 1 run
  Lengths:     1
  Values : chr16
Levels(25): chr1 chr2 chr3 chr4 chr5 chr6 chr7 chr8 chr9 chr10 chr11 ... chr15 chr16 chr17 chr18 chr19 chr20 chr21 chr22 chrX chrY chrM
```


```
> ranges
IRanges object with 1 range and 0 metadata columns:
                         start       end     width
                     <integer> <integer> <integer>
  ENSG00000140945.17  82626965  83800640   1173676
```


```
> strand
factor-Rle of length 1 with 1 run
  Lengths: 1
  Values : +
Levels(3): + - *
```

```
> name
[1] "CDH13"
```

This works -- no errors, and it launches the UCSC browser correctly:

```
targetTrack = with(targets_df, GRangesForUCSCGenome("hg19", chr, ranges, strand, name))

session <- browserSession("UCSC")
track(session, "targets") <- targetTrack
subTargetTrack <- targetTrack[1] #not actually necessary here, but you can make targetTrack a list
view <- browserView(session, subTargetTrack * -10, pack = "targets")
```

This does not work:

```
targetTrack = with(targets_df, GRangesForUCSCGenome("hg38", chr, ranges, strand, name))

session <- browserSession("UCSC")
track(session, "targets") <- targetTrack
subTargetTrack <- targetTrack[1] #not actually necessary here, but you can make targetTrack a list
view <- browserView(session, subTargetTrack * -10, pack = "targets")
```


With the following error:

```
Error in GRangesForGenome(genome, chrom = chrom, ranges = ranges, method = "UCSC", : Failed to obtain information for genome 'hg38'
11.
stop("Failed to obtain information for genome '", genome, "'")
10.
GRangesForGenome(genome, chrom = chrom, ranges = ranges, method = "UCSC", seqinfo = NULL, ...)
9.
GRangesForUCSCGenome("hg38", chr, ranges, strand, name)
8.
eval(expr, as.env(envir, enclos))
7.
eval(expr, as.env(envir, enclos))
6.
eval(expr, as.env(envir, enclos))
5.
eval(expr, envir, enclos)
4.
eval(expr, envir, enclos)
3.
safeEval(substitute(expr), data, parent.frame(), ...)
2.
with(targets_df, GRangesForUCSCGenome("hg38", chr, ranges, strand, name))
1.
with(targets_df, GRangesForUCSCGenome("hg38", chr, ranges, strand, name))
```

```
> sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /opt/intel/compilers_and_libraries_2020.0.166/linux/mkl/lib/intel64_lin/libmkl_rt.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1        stringr_1.4.0        dplyr_1.0.7          purrr_0.3.4          readr_2.0.2          tidyr_1.1.4          tibble_3.1.5        
 [8] ggplot2_3.3.5        tidyverse_1.3.1      rtracklayer_1.52.1   GenomicRanges_1.44.0 GenomeInfoDb_1.28.4  IRanges_2.26.0       S4Vectors_0.30.2    
[15] BiocGenerics_0.38.0 

loaded via a namespace (and not attached):
 [1] bitops_1.0-7                matrixStats_0.61.0          fs_1.5.0                    usethis_2.1.3               lubridate_1.8.0            
 [6] httr_1.4.2                  rprojroot_2.0.2             tools_4.1.2                 backports_1.3.0             utf8_1.2.2                 
[11] R6_2.5.1                    DBI_1.1.1                   colorspace_2.0-2            withr_2.4.2                 tidyselect_1.1.1           
[16] compiler_4.1.2              cli_3.1.0                   rvest_1.0.2                 Biobase_2.52.0              xml2_1.3.2                 
[21] desc_1.4.0                  DelayedArray_0.18.0         scales_1.1.1                Rsamtools_2.8.0             XVector_0.32.0             
[26] pkgconfig_2.0.3             sessioninfo_1.1.1           MatrixGenerics_1.4.3        dbplyr_2.1.1                rlang_0.4.12               
[31] readxl_1.3.1                rstudioapi_0.13             BiocIO_1.2.0                generics_0.1.1              jsonlite_1.7.2             
[36] BiocParallel_1.26.2         RCurl_1.98-1.5              magrittr_2.0.1              GenomeInfoDbData_1.2.6      Matrix_1.3-4               
[41] Rcpp_1.0.7                  munsell_0.5.0               fansi_0.5.0                 lifecycle_1.0.1             stringi_1.7.5              
[46] whisker_0.4                 yaml_2.2.1                  SummarizedExperiment_1.22.0 zlibbioc_1.38.0             grid_4.1.2                 
[51] crayon_1.4.1                lattice_0.20-45             Biostrings_2.60.2           haven_2.4.3                 hms_1.1.1                  
[56] knitr_1.36                  pillar_1.6.4                rjson_0.2.20                reprex_2.0.1                XML_3.99-0.8               
[61] glue_1.4.2                  BiocManager_1.30.16         modelr_0.1.8                vctrs_0.3.8                 tzdb_0.2.0                 
[66] cellranger_1.1.0            gtable_0.3.0                assertthat_0.2.1            xfun_0.27                   broom_0.7.9                
[71] restfulr_0.0.13             roxygen2_7.1.2              GenomicAlignments_1.28.0    ellipsis_0.3.2
```

**Comment by sanchit-saini** [comment]
**TLDR:** `Seqinfo(genome = "hg38")` isn't working expectedly, Which is further causing `GRangesForUCSCGenome` to fail.
As it can't get sequence information.
### POC:
```R
> suppressPackageStartupMessages(library(GenomeInfoDb))
> Seqinfo(genome="hg38")
Error in .order_seqlevels(chrom_sizes[, "chrom"]) :
  !anyNA(m32) is not TRUE
```

---
---

### Trying to recreate to see traceback
```R
> suppressPackageStartupMessages(library(rtracklayer))
> GRangesForUCSCGenome("hg38", "chr16")
Error in GRangesForGenome(genome, chrom = chrom, ranges = ranges, method = "UCSC",  :
  Failed to obtain information for genome 'hg38'

> traceback()
3: stop("Failed to obtain information for genome '", genome, "'")
2: GRangesForGenome(genome, chrom = chrom, ranges = ranges, method = "UCSC",
       seqinfo = NULL, ...)
1: GRangesForUCSCGenome("hg38", "chr16")

```

https://github.com/lawremi/rtracklayer/blob/2c4936861f62f62173097d6eeddb92936c570134/R/ranges.R#L107-L109

It's failing because sequence information is NULL, so it means `seqinfoForGenome` isn't working expectedly.

### Debugging seqinfoForGenome
```R
> rtracklayer:::seqinfoForGenome("hg38", "auto")
NULL
```

https://github.com/lawremi/rtracklayer/blob/2c4936861f62f62173097d6eeddb92936c570134/R/ranges.R#L69-L78

```R
> rtracklayer:::SeqinfoForBSGenome("hg38")
NULL
> rtracklayer:::SeqinfoForUCSCGenome("hg38")
NULL
```
### Debugging SeqinfoForBSGenome
https://github.com/lawremi/rtracklayer/blob/2c4936861f62f62173097d6eeddb92936c570134/R/ranges.R#L89-L94

```R
> rtracklayer:::BSGenomeForID("hg38")
NULL
```
https://github.com/lawremi/rtracklayer/blob/2c4936861f62f62173097d6eeddb92936c570134/R/ranges.R#L80-L87

```R
> BSgenome::getBSgenome("hg38")
Error in .stopOnAvailablePkg(genome) :
  BSgenome.Hsapiens.UCSC.hg38 package is not currently installed.
  You first need to install it, which you can do with:
      library(BiocManager)
      install("BSgenome.Hsapiens.UCSC.hg38")
```

### Debugging SeqinfoForUCSCGenome

https://github.com/lawremi/rtracklayer/blob/2c4936861f62f62173097d6eeddb92936c570134/R/ucsc.R#L133-L139

```R
> session <- browserSession("UCSC")
> genome(session) <- "hg38"
> seqinfo(session)
Error in .order_seqlevels(chrom_sizes[, "chrom"]) :
  !anyNA(m32) is not TRUE
> traceback()
10: stop(simpleError(msg, call = if (p <- sys.parent(1L)) sys.call(p)))
9: stopifnot(!anyNA(m32)) at hg38.R#36
8: .order_seqlevels(chrom_sizes[, "chrom"]) at hg38.R#67
7: GET_CHROM_SIZES(goldenPath.url = goldenPath.url)
6: .get_chrom_info_for_registered_UCSC_genome(script_path, assembled.molecules.only = assembled.molecules.only,
       map.NCBI = map.NCBI, add.ensembl.col = add.ensembl.col, goldenPath.url = goldenPath.url,
       recache = recache)
5: getChromInfoFromUCSC(genome, as.Seqinfo = TRUE)
4: .make_Seqinfo_from_genome(genome)
3: Seqinfo(genome = genome(x))
2: seqinfo(session)
1: seqinfo(session)
```

https://github.com/lawremi/rtracklayer/blob/2c4936861f62f62173097d6eeddb92936c570134/R/ucsc.R#L43-L45

```R
> Seqinfo(genome = "hg38")
Error in .order_seqlevels(chrom_sizes[, "chrom"]) :
  !anyNA(m32) is not TRUE
```

**Comment by sanchit-saini** [comment]
Opend a issue here:  https://github.com/Bioconductor/GenomeInfoDb/issues/32

**Comment by sanchit-saini** [comment]
This issue has been addressed in the latest release of Bioconductor.

Please update Bioconductor release:
```R
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install(version = "3.14")
```

**Comment by cmatKhan** [comment]
Thank you very much -- I appreciate the help!
